### PR TITLE
Create association to main waypoint when creating/updating route

### PR DESCRIPTION
**Pull request by tsauerwein**
The `main_waypoint_id` field can be set when creating or updating a route. But currently no explicit association is created between the route and this main waypoint.

### CODE REVIEW

**Comment by tsauerwein** [comment]
Pull-request attached.

**Comment by tsauerwein** [comment]
This PR is still open to review

File: c2corg_api/views/area.py [hunk 1]
```diff
@@ -47,13 +47,13 @@ def put(self):
         return self._put(Area, schema_area, after_update=update_associations)
 
 
-def insert_associations(area):
+def insert_associations(area, user_id):
```
Comment (asaunier) [inline c2corg_api/views/area.py]: Why do we need a user_id here?

Comment (tsauerwein) [inline c2corg_api/views/area.py]: `insert_associations` is called down here:
https://github.com/c2corg/v6_api/pull/177/files/8a2af2cbd823d03305c69d4e3c9611027caabf58..e447928a757b40dde6a4c763044d7cb109b4993c#diff-b906af94797c5db6c394971ac60bf7a4R218

Some methods that use the `after_add` hook require the user id.

Comment (asaunier) [inline c2corg_api/views/area.py]: OK, so you mean that ``user_id`` is passed anyway when calling the ``after_add`` hook and that the hooked methods should have it in their signature, even though not actually using it?

**Comment by asaunier** [comment]
Well, I would say that the main waypoint should first be associated to a route but indeed it's worth checking it is actually associated when setting a new main waypoint.

File: c2corg_api/views/area.py (hunk 1 above)
Comment (tsauerwein) [inline c2corg_api/views/area.py]: Exactly.

**Comment by asaunier** [comment]
Looks good to me!

**Comment by tsauerwein** [comment]
> Well, I would say that the main waypoint should first be associated to a route but indeed it's worth checking it is actually associated when setting a new main waypoint.

When updating a route, yes. But when creating a new route, it has to be done at the same time.

**Comment by tsauerwein** [comment]
Thanks for the review.

**Pull request merged**